### PR TITLE
AV-160076 Cleanup istio resources after istio is disabled

### PR DESCRIPTION
**Pull request by aaha97**
istio resources should now get cleaned up as per regular workflow if istioenabled is set to false or deleteconfig is set to true

### CODE REVIEW

File: internal/cache/controller_obj_cache.go [hunk 1]
```diff
@@ -355,7 +355,12 @@ func (c *AviObjCache) DeleteUnmarked(childCollection []string) {
 	for _, objkey := range c.SSLKeyCache.AviGetAllKeys() {
 		intf, _ := c.SSLKeyCache.AviCacheGet(objkey)
 		if obj, ok := intf.(*AviSSLCache); ok {
-			if obj.HasReference == false && obj.Name != lib.GetIstioWorkloadCertificateName() {
+			if obj.HasReference == false {
+				// if deleteConfig is false and istio is enabled, do not delete istio sslkeycert
+				if !lib.GetDeleteConfigMap() && lib.IsIstioEnabled() &&
```
Comment (akshayhavile) [inline internal/cache/controller_obj_cache.go:360]: Should we optimize this condition as `obj.Name == lib.GetIstioWorkloadCertificateName() && !lib.GetDeleteConfigMap() && lib.IsIstioEnabled()` 
Reason: With current condition, For most of the objects, first two conditions will be true (when deleteConfig is false and istio is disabled) and then last condition will be false. 
So we can avoid processing these two conditions for remaining objects

Comment (akshayhavile) [reply internal/cache/controller_obj_cache.go:360]: Similar changes at other places.

Comment (aaha97) [reply internal/cache/controller_obj_cache.go]: changed it to have the order of rarity of encounter, name > istioEnabled true > deleteConfig false

**Review by swathinsankaran — APPROVED** [review]
LGTM.

**Comment by swathinsankaran** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]

**Comment by aaha97** [comment]
build ako

**Pull request merged**